### PR TITLE
Add tag-on-create support

### PR DESCRIPTION
**Pull request by yueny2020**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-codegurureviewer-repositoryassociation/pom.xml [hunk 1]
```diff
@@ -65,6 +65,7 @@
         <dependency>
             <groupId>software.amazon.awssdk</groupId>
             <artifactId>codegurureviewer</artifactId>
+            <version>RELEASE</version>
```
Comment (vdinesh90) [inline aws-codegurureviewer-repositoryassociation/pom.xml:68]: why RELEASE?

File: aws-codegurureviewer-repositoryassociation/resource-role.yaml [hunk 2]
```diff
@@ -29,6 +29,8 @@ Resources:
                 - "codeguru-reviewer:AssociateRepository"
                 - "codeguru-reviewer:DescribeRepositoryAssociation"
                 - "codeguru-reviewer:DisassociateRepository"
+                - "codeguru-reviewer:ListTagsForResource"
+                - "codeguru-reviewer:TagResource"
```
Comment (vdinesh90) [inline aws-codegurureviewer-repositoryassociation/resource-role.yaml:33]: how about removing tags ? would the tags get removed as part of disassociate?

File: aws-codegurureviewer-repositoryassociation/src/main/java/software/amazon/codegurureviewer/repositoryassociation/Translator.java [hunk 3]
```diff
@@ -55,6 +60,14 @@ static AssociateRepositoryRequest translateToAssociateRepositoryRequest(final Re
...
 
+        final Optional<Map<String, String>> tags = getTagsFromModel(model);
+        if(tags.isPresent()) {
+            return AssociateRepositoryRequest.builder()
+                    .repository(repository)
+                    .tags(tags.get())
+                    .build();
+        }
+
         return AssociateRepositoryRequest.builder().repository(repository).build();
```
Comment (vdinesh90) [inline aws-codegurureviewer-repositoryassociation/src/main/java/software/amazon/codegurureviewer/repositoryassociation/Translator.java:71]: `tagsOptional.map( "with tags").orElse("without tags")`

this is much more readable with the usage of Optional.

File: aws-codegurureviewer-repositoryassociation/src/main/java/software/amazon/codegurureviewer/repositoryassociation/Translator.java [hunk 4]
```diff
@@ -79,7 +92,13 @@ static ResourceModel translateFromReadResponse(final DescribeRepositoryAssociati
         if(!providerType.equals(ProviderType.CODE_COMMIT)) {
             resourceModelBuilder.owner(awsResponse.repositoryAssociation().owner());
         }
-
+        Map<String, String> tags = awsResponse.tags();
```
Comment (vdinesh90) [inline aws-codegurureviewer-repositoryassociation/src/main/java/software/amazon/codegurureviewer/repositoryassociation/Translator.java:95]: Can tags field be null in awsResponse?

File: aws-codegurureviewer-repositoryassociation/resource-role.yaml (hunk 2 above)
Comment (yueny2020) [reply aws-codegurureviewer-repositoryassociation/resource-role.yaml:33]: Tags will not be remove when disassociation. Tags will be removed when the associationId automatically deletes after in disassociated status of 90 days.

File: aws-codegurureviewer-repositoryassociation/src/main/java/software/amazon/codegurureviewer/repositoryassociation/Translator.java (hunk 4 above)
Comment (yueny2020) [reply aws-codegurureviewer-repositoryassociation/src/main/java/software/amazon/codegurureviewer/repositoryassociation/Translator.java:95]: We return `Tags` field for DescribeRepositoryAssociationResponse. This field can be empty but not null.

File: aws-codegurureviewer-repositoryassociation/pom.xml (hunk 1 above)
Comment (yueny2020) [reply aws-codegurureviewer-repositoryassociation/pom.xml:68]: The dependencyManagement specifies the version of software.amazon.awssdk to be 2.15.0. The tag feature of codegurureviewer is released in version 2.15.33.

File: aws-codegurureviewer-repositoryassociation/src/main/java/software/amazon/codegurureviewer/repositoryassociation/Translator.java (hunk 3 above)
Comment (yueny2020) [reply aws-codegurureviewer-repositoryassociation/src/main/java/software/amazon/codegurureviewer/repositoryassociation/Translator.java:71]: I agree Optional.map().orElse() is a more readable usage. But I think it is not quite fit into the above situation.

File: aws-codegurureviewer-repositoryassociation/pom.xml (hunk 1 above)
Comment (vdinesh90) [reply aws-codegurureviewer-repositoryassociation/pom.xml:68]: lets pick a version that's latest, i'm not sure if RELEASE picks the latest version.

**Review by vdinesh90 — APPROVED** [review]

**Pull request merged**